static/js/stream_ui_updates.js

```diff
@@ -178,6 +180,36 @@ export function update_permissions_banner(sub) {
     $settings.find(".stream-settings-tip-container").html(rendered_tip);
 }
 
+export function update_notifications_stream_in_settings() {
+    // if stream creation form is not open, return
+    if (!hash_util.is_create_new_stream_narrow()) {
+        return;
+    }
+
+    // if announcement is turned off, hide the checkbox
+    if (!stream_data.realm_has_notifications_stream()) {
+        $("#announce-new-stream .checkbox").hide();
+        return;
+    }
+
+    if (stream_data.realm_has_notifications_stream()) {
+        $("#announce-new-stream .checkbox").show();
+    }
+
+    const notifications_stream = stream_data.get_notifications_stream();
+    const notifications_stream_sub = stream_data.get_sub_by_name(notifications_stream);
+    const stream_privacy_symbol_html = render_stream_privacy({
```

Comment:
This gives error when user does not have access to the steam and we need to hide the option in that case as mentioned in the issue. Would be better if you divide the work into two commits - One for reordering the contents and to add the `Stream permissions` heading. And other for the work related to `Announce stream` option like adding stream name to the label and hiding the option when user does not have access to the stream.

Reply:
@sahil839  This part of code hides the option and return from the function. Isn't this somethething you are looking for?
```// if announcement is turned off, hide the checkbox
    if (!stream_data.realm_has_notifications_stream()) {
        $("#announce-new-stream .checkbox").hide();
        return;
    }
```

Reply:
This hides when there is no notification stream set, but there can be case when notification stream is set to a stream which user does not have access to, like an unsubscribed private stream, so it gives error for such case. See the first point under second checkbox in the issue for expected behavior.